components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/PasswordGrantHandler.java

```diff
@@ -136,7 +137,17 @@ private boolean validateUserCredentials(OAuth2AccessTokenReqDTO tokenReq) throws
                 throw new IdentityOAuth2Exception("Authentication failed for " + tokenReq.getResourceOwnerUsername());
             }
         } catch (UserStoreException e) {
-            throw new IdentityOAuth2Exception("Error while authenticating user from user store");
+            String message;
+            IdentityOAuth2Exception identityOAuth2Exception;
+            // Set error code to message if available.
+            if (e.getCause() instanceof IdentityException && ((IdentityException) e.getCause()).getErrorCode() != null){
+                message = ((IdentityException) e.getCause()).getErrorCode() + " " + e.getMessage();
+                identityOAuth2Exception = new IdentityOAuth2Exception(message, e);
+            } else {
+                message = e.getMessage();
+                identityOAuth2Exception = new IdentityOAuth2Exception(message, e);
+            }
```

Comment:
can bring identityOAuth2Exception = new IdentityOAuth2Exception(message, e); out of the if condition